cpp/dolfinx/common/DolfinXException.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <stdexcept>
+
+class DolfinXException : public std::runtime_error
```

Comment:
I would propose a namespaced ``runtime_error`` exception, e.g.

``dolfinx::runtime_error``

I think this should be straightforward to do with search and replace.

Reply:
I added the suggestion